db/sql/05_msar.sql

```diff
@@ -5627,6 +5627,49 @@ END;
 $$ LANGUAGE plpgsql;
 
 
+CREATE OR REPLACE FUNCTION msar.list_by_record_summaries(
+  tab_id oid,
+  limit_ integer,
+  offset_ integer,
+  search_ text DEFAULT NULL,
+  table_record_summary_templates jsonb DEFAULT NULL
+) RETURNS jsonb
+LANGUAGE plpgsql STABLE
+AS $$
+DECLARE
+  search_where_clause text := '';
+  final_sql text;
+  result_json jsonb;
+BEGIN
+  IF search_ IS NOT NULL AND search_ <> '' THEN
+    search_where_clause := format(' WHERE summary ILIKE %L', '%'||search_||'%');
+  END IF;
+
+  final_sql := format(
+    $q$
+    WITH
+      all_record_summaries AS ( %1$s ),
+      filtered AS ( SELECT * FROM all_record_summaries %2$s ),
+      sorted AS ( SELECT * FROM filtered ORDER BY summary LIMIT %3$s OFFSET %4$s )
+    SELECT
+      json_build_object(
+        'count', (SELECT count(*) FROM filtered),
```

Comment:
Optional: Unless there's a good reason, I'd prefer this be a CTE for consistency.